### PR TITLE
Added pending/history page for settings

### LINKED ISSUE #135: Dev: Governance Setting – Pending Requests/History
# Background
We recently introduced a Governance Setting page for admin to configure various settings (see https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/issues/90, https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/issues/116, https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/issues/124). The admin is able to make various setting policy requests, but currently there is no way to vote on them and take effect once the required voting threshold is met. This prevents the admin from actually applying these changes, leaving them in an infinite pending state.

This ticket focuses on implementing a "Pending Requests" component within the Settings page, which would allow users with voting permissions to see all pending setting policy requests and vote on them, which ultimately will enable the teams to apply the changes.

# User Story
As an admin with vote permissions, I want the ability to vote on pending setting requests so that the policy changes we want take effect.

# Acceptance Criteria

## 1. Requests Configuration Section
- [x] A new "Requests" section is added to the Treasury Settings left sidebar. It is the first page that users see when they click on the global "Settings" tab.
- [x] The right side of the Requests section is broken into two tabs, displaying the total number of requests next to their titles
  - [x] Pending Requests
  - [x] History
  - [x] There is pagination at the bottom to handle a large number of requests (e.g. greater than 20).
  - [ ] Sorting options are available to organize the requests by clicking on the column header.
- [x] Permissions: 
  - [x] All users can view the new Requests section, but cannot take any action unless they have permissions
  - [x] Users with Vote permissions can approve or reject pending request
  - [x] Users with Create Request can delete pending requests ** NOTE – need to confirm this with @Megha-Dev-19 

## 2. Pending Requests Tab
- [x] The "Pending Requests" tab displays a list of all pending Settings requests, including the following columns
  - [x] ID: The ID of the pending request (default sort, ascending order from 01)
  - [x] Created Date: The date the pending request was created (Note: sorted by default by most recent at the top)
  - [x] Title: Shows the pending request type
  - [x] Created By: Shows the username of the person who created this request
  - [x] Required Votes: Shows how many votes are required
  - [x] Votes: Shows voting distribution
  - [x] Approvers: Shows who approved the request or rejected
  - [x] Expiring Date: Shows when the request expires
  - [x] Actions: Buttons for approving, rejecting, or viewing details
- [x] Requests require the threshold set by the voting thresholds policy settings

## 3. History Tab
- [x] The "History" tab displays a list of all completed or expired Settings requests, including the following columns
  - [x] ID: The ID of the pending request (default sort, ascending order from 01)
  - [x] Created Date: The date the pending request was created (Note: sorted by default by most recent at the top)
  - [x] Status: The status of the request (e.g. Rejected, Expired, Approved)
  - [x] Title: Shows the pending request type
  - [x] Created By: Shows the username of the person who created this request
  - [x] Approvers: Shows who approved or rejected

## 4. Processing Requests
- [x] When a user clicks Approve, they see a confirmation. Once they approve, they will see "You approved" message.
- [x] Once the voting threshold is met and the decision is to approve, the policy takes effect and the pending request moves to history tab with "Approved" status
 - [x] Once the voting threshold is met and the decision is to reject, the policy takes effect and the pending request moves to history tab with "Rejected" status
- [x] If the request expires before the voting threshold is met, the pending requests moves to history tab with "Expired" status

# Reference
See Settings [Figma prototype](https://www.figma.com/proto/hNg69dceqgi7yMYtovrbYc/%F0%9F%8F%A6-Multisig-Treasury-Application?page-id=4221%3A16322&node-id=4673-5484&node-type=canvas&viewport=-8487%2C-7592%2C0.85&t=sL0p1YbDPR7MrfNF-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=848%3A12117), specifically the Requests tab

### PR DESCRIPTION
**Pull request by Megha-Dev-19**
Resolves #135 
[video.webm](https://github.com/user-attachments/assets/259fbe67-0b30-4cb7-a1e6-19d1e8dce5d8)
[video.webm](https://github.com/user-attachments/assets/8a43ca57-a544-4613-aa73-c5ea5ce7dbaa)
[video.webm](https://github.com/user-attachments/assets/5bd52eb9-6cd7-4231-9a96-4129d0671c1b)
[video.webm](https://github.com/user-attachments/assets/8eec8887-a08b-4250-aea8-1dfd34b529cd)

### CODE REVIEW

**Comment by race-of-sloths** [comment]
@Megha-Dev-19 Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!


[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/Megha-Dev-19?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/172&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/Megha-Dev-19?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/172&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/Megha-Dev-19)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/Megha-Dev-19)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by petersalomonsen — APPROVED** [review]
@race-of-sloths score 13

**Pull request merged**
